t/t1002-mf-priority-small-no-tie.t

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+test_description='Test multi-factor priority plugin order with no ties'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SUBMIT_AS=${FLUX_BUILD_DIR}/t/scripts/submit_as.py
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+	flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'check that mf_priority plugin is loaded' '
+	flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'create a group of users with unique fairshare values' '
+	cat <<-EOF >fake_small_no_tie.py
+	import flux
+	import pwd
+	import getpass
+
+	username = getpass.getuser()
+	userid = pwd.getpwnam(username).pw_uid
+	# create a JSON payload
+	data = {"userid": "5011", "bank": "account1", "fairshare": "0.285714"}
+	flux.Flux().rpc("job-manager.mf_priority.rec_update", data).get()
```

Comment:
Ah, okay, thanks for this suggestion. Maybe I should try to include it in this PR. I'll play around with it today and see if I can get one script that can just be re-used across these tests. I also noticed some unneeded code in there too:

```python
import pwd
import getpass

username = getpass.getuser()
userid = pwd.getpwnam(username).pw_uid
```

I don't think I'll need these lines, so I should remove those too.

Reply:
The way I think about it is that you'll probably want to add a lot of these types of tests, so you should make it as easy as possible to add new scenarios. We have some tests in flux-core where all you have to do is drop an "input" file and "expected output" file in a directory and it will automatically be picked up by the test harness. Not sure you have to go that far at this time, but it is useful to be able to quickly devise and implement tests in a testsuite.

Reply:
Absolutely. Just from playing around with it a little bit, I was able to create a pretty simple Python script with just a few lines:

```python
h = flux.Flux()

with open(sys.argv[1]) as data_file:
    data = json.load(data_file)

for user in data["users"]:
    h.rpc("job-manager.mf_priority.rec_update", user).get()
```

that can read in a `.json` file containing an array of user data with `flux python send_payload.py <name of .json file>`. I just pushed it up as an additional commit. I also added a `--fixup` commit that edits the sharness tests to make use of this new helper script. I am still creating the `.json` file in the test file itself, but if you think it would be better to create an `input` directory where these files are placed, that works fine with me as well. 👍

Reply:
Creating the file in the test seems fine for now. Later if you want to have tens or hundreds of tests you might want to have a directory of input :wink:

Reply:
That makes sense to me. I just pushed an update to squash that `fixup` commit.